### PR TITLE
Close #3286 Adding a better exposed filter plugin for views.

### LINKED ISSUE #3286: Add ViewsExposedForm Plugin to override and enhance views exposed form filters/sorts/pagers
### Is your feature request related to a problem? Please describe.

Site builders are unable to utilize Better Exposed Filters to provide end users with functionality that displays the number of active filters are selected. Plus, site builders would like  to adjust the reset button's position on an exposed form.

Additionally, users who rely on keyboard navigation or screen readers face challenges when accessing exposed filters in views due to the lack of a direct method for navigating to these filters. This limitation hampers the site's accessibility and usability for individuals depending on keyboard navigation or assistive technologies.

### Describe the solution you'd like

The proposal is to introduce a new module named `az_finder`, which includes a plugin that extends the Better Exposed Filters (BEF) plugin. This extension aims to incorporate a configurable skip link at the top of views with exposed filters, thereby facilitating direct and improved access to the filters and enhancing overall site navigation and accessibility. Additionally, the plugin will offer the following configurable features to further refine user interaction with exposed filters:

- `reset_button_position`: Allows the positioning of the reset button to be altered, with "top" as a configurable option to place the reset button above the filters, enhancing accessibility and visibility.
- `min_text_length`: Sets a minimum character count that text fields must reach before being considered active, preventing premature form submissions and ensuring more intentional filter usage.
- `reset_button_counter`: Introduces a counter next to the reset button to indicate the number of active filters, aiding users in tracking their filter usage.
- `orientation`: Enables setting the form orientation to vertical, facilitating a more structured and accessible layout for users navigating with keyboards or screen readers.

These settings will be easily configurable, providing flexibility to accommodate different use cases and preferences while prioritizing accessibility.

### Describe alternatives you've considered

Two alternatives were evaluated:
1. **Enhancing Tab Order**: Modifying the site's tab order to expedite access to exposed filters. This method falls short in providing the directness offered by a skip link.
2. **ARIA Landmarks**: Using ARIA landmarks to improve navigation. Although useful, this approach lacks the straightforward action of a skip link, which directly benefits those seeking to utilize the filters.

### Roles and Permissions considerations

Different user roles will have varying levels of access and configuration capabilities for the skip link feature:
- **Anonymous User**: Granted access to the skip link when exposed filters are visible and not restricted by permissions.
- **Authenticated User and Content Editors/Administrators**: Similar access to anonymous users, with potential additional filtering options based on user roles.
- **Administrator**: Full authority to configure, enable, or disable the skip link feature, including setting permissions for other roles regarding configuration capabilities.

## Additional Context

Implementing a skip link that allows users to jump directly to exposed filters is a recognized best practice for web accessibility. This feature would significantly improve the user experience for those utilizing keyboard navigation or assistive technologies, aligning the site more closely with accessibility standards.

### PR DESCRIPTION
**Pull request by trackleft**
## Description

This Pull Request introduces a new experimental module with a `ViewsExposedForm` plugin. This plugin, dependent on the `better_exposed_filters` contrib module and views, enhances the flexibility and functionality of views with exposed forms. It offers a variety of settings to customize the user interface for filtering, sorting, or paging actions within a view.

The plugin can be activated within the Advanced section of the Exposed form settings area of any view utilizing exposed forms. The addition of this plugin introduces several new settings, allowing for a tailored fit to specific needs:
~~min_text_length: 1~~

```yaml
reset_button_position: top
reset_button_counter: true
orientation: vertical
skip_link: true
skip_link_text: 'Skip to search and filter'
skip_link_id: search-filter
```

- `reset_button_position`: Changes the reset button's position (top of the form is the default).
~~- `min_text_length`: Sets a character minimum for text fields to be considered active and prevent form submission if not met.~~
- `reset_button_counter`: Adds a counter to the reset button indicating the number of active filters.
- `orientation`: Adds a class to set the form's orientation (horizontal or vertical), note: no CSS is included in this PR.
- `skip_link`: Adds a skip link to the top of the view display.
- `skip_link_text`: Customizes the skip link text.
- `skip_link_id`: Sets the ID for the filters, with a sensible default provided.

The `az_finder` view within `az_demo` has been updated to utilize this new plugin, serving as a practical example and aiding in acceptance testing and QA.

## Implementation notes:

1. Extending the reset button functionality was challenging due to Drupal's limitations on HTML within submit buttons. A workaround involved altering Drupal's form render element to accommodate our needs, which might require further refinement.

2. The skip link functionality introduces an accessibility feature, but it requires careful implementation on the site builder's part to ensure the skip link points to the correct target element.

3. I've rearranged the configuration form quite a bit and am massaging form values prior to saving.
Plugin Settings Form:
image

4. The reset button does not work with the select all/none functionality, I think this is due to [better_exposed_filters choosing to halt event propagation ](https://git.drupalcode.org/project/better_exposed_filters/-/blob/6.0.x/js/bef_select_all_none.js?ref_type=heads#L24). 

5. I separated out the javascript into two libraries, one for the reset button counter and one for the reset button.

## Related issues
- Close #3286 
- Blocks #2998 

### Issues discovered while building this PR
- #3296 
- #3293 

### Pull requests for the found issues:
- #3298 
- #3294 

## How to test

1. Navigate to the Views UI and clone the `az_finder` view. Assign a unique name to your cloned view, such as `az_finder_chris`, to personalize it.
2. Explore the new settings available through the `ViewsExposedForm` plugin by accessing the Exposed form settings area within the Advanced section of your cloned view's configuration.
3. Methodically adjust the newly introduced settings, such as `reset_button_position`, ~~`min_text_length`~~, `reset_button_counter`, `orientation`, `skip_link`, `skip_link_text`, and `skip_link_id`. After each change, save your view.
4. Ensure to clear your Drupal cache to see the effects of your modifications reflected on the site.
5. Experiment with different combinations of settings to understand how they interact and the range of customization they offer for exposed forms in views.

This hands-on approach will allow reviewers to fully grasp the capabilities and flexibility introduced by the new plugin, ensuring a thorough evaluation of its functionalities.

## Action Items
- [ ] Determine defaults for all settings.
- [ ] Fix for Select All/None
- [ ] Is the Module Description OK?
- [ ] Add option for scrolling to top of view after clicking the submit button (don't enable for auto submit)
- [x] Remove min text length settings because it doesn't work.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [x] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 1]
```diff
@@ -1560,22 +1602,29 @@ display:
                   sort_options: false
                   rewrite:
                     filter_rewrite_values: ''
-                  collapsible: true
+                  collapsible: false
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:1605]:
```suggestion
                  collapsible: true
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 2]
```diff
@@ -1560,22 +1602,29 @@ display:
...
                   is_secondary: false
-                select_all_none: true
-                select_all_none_nested: true
+                select_all_none: false
+                select_all_none_nested: false
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:1608]:
```suggestion
                select_all_none: true
                select_all_none_nested: true
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 3]
```diff
@@ -388,6 +388,11 @@ display:
               secondary_label: 'Advanced options'
               secondary_open: false
               reset_button_always_show: true
+              skip_link: 0
+              skip_link_settings:
+                skip_link_text: 'Skip to search and filter'
+                skip_link_id: search-filter
+              orientation: horizontal
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:395]:
```suggestion
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 4]
```diff
@@ -401,22 +401,18 @@ display:
...
-                select_all_none_nested: true
-                display_inline: false
-              tid:
-                plugin_id: bef
-                advanced:
-                  sort_options: false
-                  rewrite:
-                    filter_rewrite_values: ''
-                  collapsible: true
+                  collapsible: false
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:404]:
```suggestion
                  collapsible: true
                  is_secondary: false
                select_all_none: true
                select_all_none_nested: true
                display_inline: false
              tid:
                plugin_id: bef
                advanced:
                  sort_options: false
                  rewrite:
                    filter_rewrite_values: ''
                  collapsible: true```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 5]
```diff
@@ -588,6 +595,11 @@ display:
               secondary_label: 'Advanced options'
               secondary_open: false
               reset_button_always_show: true
+              skip_link: 0
+              skip_link_settings:
+                skip_link_text: 'Skip to search and filter'
+                skip_link_id: search-filter
+              orientation: horizontal
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:602]:
```suggestion
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 6]
```diff
@@ -864,6 +878,11 @@ display:
...
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:885]:
```suggestion
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 7]
```diff
@@ -875,7 +889,7 @@ display:
                 plugin_id: default
                 advanced:
                   placeholder_text: ''
-                  collapsible: true
+                  collapsible: false
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:892]:
```suggestion
                  collapsible: true
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 8]
```diff
@@ -889,11 +903,18 @@ display:
                   sort_options: false
                   rewrite:
                     filter_rewrite_values: ''
-                  collapsible: true
+                  collapsible: false
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:906]:
```suggestion
                  collapsible: true
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 9]
```diff
@@ -891,9 +905,16 @@ display:
                     filter_rewrite_values: ''
                   collapsible: true
                   is_secondary: false
-                select_all_none: true
-                select_all_none_nested: true
+                select_all_none: false
+                select_all_none_nested: false
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:909]:
```suggestion
                select_all_none: true
                select_all_none_nested: true
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 10]
```diff
@@ -1265,6 +1286,11 @@ display:
               secondary_label: 'Advanced options'
               secondary_open: false
               reset_button_always_show: true
+              skip_link: 0
+              skip_link_settings:
+                skip_link_text: 'Skip to search and filter'
+                skip_link_id: search-filter
+              orientation: horizontal
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:1293]:
```suggestion
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 11]
```diff
@@ -1280,9 +1301,16 @@ display:
                     filter_rewrite_values: ''
                   collapsible: false
                   is_secondary: false
-                select_all_none: true
-                select_all_none_nested: true
+                select_all_none: false
+                select_all_none_nested: false
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:1305]:
```suggestion
                select_all_none: true
                select_all_none_nested: true
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 12]
```diff
@@ -1571,11 +1604,18 @@ display:
...
                   rewrite:
                     filter_rewrite_values: ''
-                  collapsible: true
-                  is_secondary: true
-                select_all_none: true
-                select_all_none_nested: true
+                  collapsible: false
+                  is_secondary: false
+                select_all_none: false
+                select_all_none_nested: false
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:1610]:
```suggestion
                  collapsible: true
                  is_secondary: true
                select_all_none: true
                select_all_none_nested: true
```

File: modules/custom/az_demo/config/install/views.view.az_finder.yml [hunk 13]
```diff
@@ -1547,6 +1575,11 @@ display:
               secondary_label: 'Advanced options'
               secondary_open: false
               reset_button_always_show: true
+              skip_link: 1
+              skip_link_settings:
+                skip_link_text: 'Skip to search and filter'
+                skip_link_id: search-filter
+              orientation: horizontal
```
Comment (trackleft) [inline modules/custom/az_demo/config/install/views.view.az_finder.yml:1582]:
```suggestion
```

**Comment by danahertzberg** [comment]
This is looking amazing! Here's some feedback (not sure if you're ready for it yet).

1. I got this error on the page when logged in as an admin.
   Screenshot 2024-03-25 at 11 54 15 AM

2. Using the search, then clicking the reset button doesn't reload all results. It only clears the count number. Works great on the checkboxes!

   2a. PS - I LOVE the check all checkbox functionality for parents. We'll need that for future updates for sure.

3. Question. How does the skip link get added to a page? Is it just associated with the block no matter where the block is?

File: modules/custom/az_finder/az_finder.libraries.yml [hunk 14]
```diff
@@ -0,0 +1,19 @@
+active-filter-count:
+  css:
+    component:
+      css/active-filter-count.css: {}
```
Comment (trackleft) [inline modules/custom/az_finder/az_finder.libraries.yml:4]:
```suggestion
```

**Comment by trackleft** [comment]
Addressing the first error with [`8a0575f` (#3289)](https://github.com/az-digital/az_quickstart/pull/3289/commits/8a0575ffdb507727a56563be7684cd371283e80d)

**Comment by danahertzberg** [comment]
Think it's worth considering a follow up ticket to load at the top of the page after submitting the filters? (Maybe only with the submit button enabled? Wouldn't be ideal for autosubmit)

**Comment by danahertzberg** [comment]
The select nested checkboxes feature isn't working as I would expect. I'm expecting it to select everything top-down. But when I select some children, it sometimes selects the parent too. This would also be a reasonable follow up.

**Comment by danahertzberg** [comment]
What does this do? I think I gather the correct assumption, but I didn't see anything different on the front end between the two selections.
Screenshot 2024-03-25 at 4 25 50 PM

**Comment by trackleft** [comment]
> What does this do? I think I gather the correct assumption, but I didn't see anything different on the front end between the two selections. Screenshot 2024-03-25 at 4 25 50 PM

This is meant to add classes to the exposed form like this
```
    if ($options['orientation'] === 'vertical') {
      $form['#attributes']['class'][] = 'az-bef-vertical';
    }
    if ($options['orientation'] === 'horizontal') {
      $form['#attributes']['class'][] = 'az-bef-horizontal';
    }
    ```

**Comment by trackleft** [comment]
> The select nested checkboxes feature isn't working as I would expect. I'm expecting it to select everything top-down. But when I select some children, it sometimes selects the parent too. This would also be a reasonable follow up.

The better exposed filter module handles this feature, if there is a single child it selects the parent, this is not consistent with how many of us think about taxonomy, since we typically think of taxonomy terms as autonomous entities without hierarchy in mind.  

While this feature does look strange, I couldn't tell if it actually affects the filtered result.

File: modules/custom/az_finder/az_finder.module [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+/**
+ * Implements hook_preprocess_HOOK() for views_view.
+ */
+function az_finder_preprocess_views_view(&$variables) {
+  $view = $variables['view'];
+
+  if (empty($view->display_handler->options['exposed_form']['type']) ||
+      $view->display_handler->options['exposed_form']['type'] !== 'az_better_exposed_filters') {
```
Comment (trackleft) [inline modules/custom/az_finder/az_finder.module:15]:
```suggestion
    $view->display_handler->options['exposed_form']['type'] !== 'az_better_exposed_filters') {
```

File: modules/custom/az_finder/az_finder.libraries.yml [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+    - core/drupal
+    - core/drupalSettings
+active-filter-reset:
+  js:
+    js/active-filter-reset.js:
+      - type: module
+      - minified: false
+  dependencies:
+    - core/drupal
+    - core/drupalSettings
```
Comment (joeparsons) [inline modules/custom/az_finder/az_finder.libraries.yml:16]: Very picky nitpick:  Looks like `active-filter-reset.js` isn't actually using `drupalSettings`/`settings`.
```suggestion
```

File: modules/custom/az_finder/az_finder.info.yml [hunk 17]
```diff
@@ -0,0 +1,7 @@
+name: 'Quickstart Finder'
+type: module
+description: 'A collection of plugins for building a quickstart finder.'
```
Comment (joeparsons) [inline modules/custom/az_finder/az_finder.info.yml:3]:
```suggestion
description: 'A collection of plugins for building finder views on Quickstart sites.'
```

File: modules/custom/az_finder/az_finder.libraries.yml (hunk 16 above)
Comment (trackleft) [reply modules/custom/az_finder/az_finder.libraries.yml:16]: I could swear I checked that, thanks

File: modules/custom/az_finder/js/active-filter-count.es6.js [hunk 18]
```diff
@@ -0,0 +1,70 @@
+((drupalSettings, Drupal) => {
```
Comment (trackleft) [inline modules/custom/az_finder/js/active-filter-count.es6.js:1]:
```suggestion
/**
 * @file
 * active-filter-count.es6.js
 *
 * This file contains the JavaScript needed to count active filters in
 * exposed filter forms.
 */
((drupalSettings, Drupal) => {
```

File: modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php [hunk 19]
```diff
@@ -0,0 +1,226 @@
...
+        'top' => $this->t('Top'),
+        'bottom' => $this->t('Bottom'),
+      ],
+      '#default_value' => $this->options['reset_button_position'] ?? 'bottom',
+      '#description' => $this->t('Select where to place the reset button in the form.'),
+    ];
+    $form['bef']['general']['reset_button_settings']['reset_button_counter'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Show active filter counter'),
+      '#description' => $this->t('Show a counter of active filters next to the reset button.'),
```
Comment (danahertzberg) [inline modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:149]:
```suggestion
      '#description' => $this->t('Show a counter of active filters within the reset button next to the text.'),
```

File: modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php [hunk 20]
```diff
@@ -0,0 +1,226 @@
...
+      unset($form['bef']['general']['reset_button_label']);
+    }
+    $form['bef']['general']['reset_button_settings']['reset_button_position'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Reset Button Position'),
+      '#options' => [
+        'top' => $this->t('Top'),
+        'bottom' => $this->t('Bottom'),
+      ],
+      '#default_value' => $this->options['reset_button_position'] ?? 'bottom',
```
Comment (danahertzberg) [inline modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:143]:
```suggestion
      '#default_value' => $this->options['reset_button_position'] ?? 'top',
```

File: modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php [hunk 21]
```diff
@@ -0,0 +1,226 @@
...
+    ];
+    $form['bef']['general']['reset_button_settings']['reset_button_counter'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Show active filter counter'),
+      '#description' => $this->t('Show a counter of active filters next to the reset button.'),
+      '#default_value' => $this->options['reset_button_counter'] ?? FALSE,
+    ];
+    $form['bef']['general']['skip_link'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Add skip link to the top of the view'),
```
Comment (danahertzberg) [inline modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:154]:
```suggestion
      '#title' => $this->t('Add skip link to the top of the view results'),
```

File: modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php [hunk 22]
```diff
@@ -0,0 +1,226 @@
...
+      '#description' => $this->t('Add a skip link to the top of the view to allow keyboard users to skip to the search and filter form.'),
```
Comment (danahertzberg) [inline modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:155]:
```suggestion
      '#description' => $this->t('Add a skip link to the top of the view results to allow keyboard users to skip to the search and filter form.'),
```

File: modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php [hunk 23]
```diff
@@ -0,0 +1,226 @@
...
+    ];
+    $form['bef']['general']['orientation'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Orientation'),
+      '#description' => $this->t('The orientation of the filters within the exposed form.'),
+      '#options' => [
+        'horizontal' => $this->t('Horizontal'),
+        'vertical' => $this->t('Vertical'),
+      ],
+      '#default_value' => $this->options['orientation'] ?? FALSE,
```
Comment (danahertzberg) [inline modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:189]: Looking to have the default orientation be vertical.

```suggestion
      '#default_value' => $this->options['orientation'] ?? TRUE,
```

File: modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php [hunk 24]
```diff
@@ -0,0 +1,226 @@
...
+      '#states' => [
+        'visible' => [
+          ':input[name="exposed_form_options[reset_button]"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+    if (isset($form['bef']['general']['reset_button_always_show'])) {
+      $form['bef']['general']['reset_button_settings']['reset_button_always_show'] = $form['bef']['general']['reset_button_always_show'];
+      unset($form['bef']['general']['reset_button_always_show']);
+    }
```
Comment (danahertzberg) [inline modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:131]: This does not show the button if unchecked and options on the front-end are selected.

File: modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php [hunk 25]
```diff
@@ -0,0 +1,226 @@
...
+      '#type' => 'checkbox',
+      '#title' => $this->t('Show active filter counter'),
+      '#description' => $this->t('Show a counter of active filters next to the reset button.'),
+      '#default_value' => $this->options['reset_button_counter'] ?? FALSE,
+    ];
+    $form['bef']['general']['skip_link'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Add skip link to the top of the view'),
+      '#description' => $this->t('Add a skip link to the top of the view to allow keyboard users to skip to the search and filter form.'),
+      '#default_value' => $this->options['skip_link'] ?? FALSE,
```
Comment (danahertzberg) [inline modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:156]:
```suggestion
      '#default_value' => $this->options['skip_link'] ?? TRUE,
```

File: modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php (hunk 24 above)
Comment (danahertzberg) [reply modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:131]: Also unable to check the "always show" option. Saved checkbox selection doesn't save.

Comment (trackleft) [reply modules/custom/az_finder/src/Plugin/views/exposed_form/QuickstartExposedFilters.php:131]: Fixed, I think. 🙇‍♂️

**Review by danahertzberg — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**